homeassistant/components/homekit_controller/select.py

```diff
@@ -17,14 +17,15 @@
     0: "home",
     1: "sleep",
     2: "away",
+    3: "vacation",
 }
 _ECOBEE_MODE_TO_NUMBERS = {v: k for (k, v) in _ECOBEE_MODE_TO_TEXT.items()}
 
 
 class EcobeeModeSelect(CharacteristicEntity, SelectEntity):
     """Represents a ecobee mode select entity."""
 
-    _attr_options = ["home", "sleep", "away"]
+    _attr_options = list(_ECOBEE_MODE_TO_TEXT.values())
```

Comment:
Since vacation mode can't be set via HomeKit, this is really only useful for displaying the current mode (when it works). In order to display vacation mode, this line needs to include "vacation". But that also makes it an option to be selected. Is there a way to make it a display-only option?

Reply:
The following lines added to `async_select_option` in this file would solve that, but it feels pretty kludgy.

```
        if option_int == 3:
            raise ValueError(f"Option {option} not valid for {self.entity_id}")
```